### PR TITLE
Ingress Changes

### PR DESCRIPTION
**Pull request by eloycoto**
I was troubleshooting `nginx-ingress` and the TLS passthrough option on Minikube was a bit painful to set. So, for now, I switched the ingress to the Openshift router, where the TCP load-balancing works as expected and things are easier to debug. 

To install: 

```
make install-router
make deploy IMG=quay.io/project-flotta/flotta-operator TARGET=k8s
```

### CODE REVIEW

File: config/network/route.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ metadata:
   name: controller-manager
   namespace: system
 spec:
+  host: REPLACE_HOST
```
Comment (masayag) [inline config/network/route.yaml:10]: iinw, in OCP this value is decided by the ingress-operator (service-name-namespace-name-cluster-name)

Comment (eloycoto) [reply config/network/route.yaml:10]: but that value should be in alignment with the flotta/CA certs.

File: .github/workflows/kind.yaml [hunk 2]
```diff
@@ -22,35 +22,32 @@ jobs:
...
           kubectl wait --for=condition=Ready pods --all -n kube-system
+          make install-router
 
           # Deploy flotta operator
           sed -i 's/LOG_LEVEL=info/LOG_LEVEL=debug/g' config/manager/controller_manager_config.properties
           make build
           IMG=flotta-operator:latest make docker-build
           kind load docker-image flotta-operator:latest
-          timeout 2m make deploy IMG=flotta-operator TARGET=kind || retVal=$?
+          timeout 2m make deploy IMG=flotta-operator TARGET=k8s || retVal=$?
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply .github/workflows/kind.yaml:46]: this PR is marked as WIP, not yet finished ;-)

File: config/network/route.yaml (hunk 1 above)
Comment (masayag) [reply config/network/route.yaml:10]: so in that case, we should pick the name when creating the cert based on expected host name, no?

Comment (eloycoto) [reply config/network/route.yaml:10]: yeah, it's a parameter on the operator config, default is project-flotta.io and 127.0.0.1

**Comment by gciavarrini** [comment]
No need to update the documentation?

**Comment by eloycoto** [comment]
> No need to update the documentation?

Just a follow-up PR in docs as soon this is finished :-)

**Review by gciavarrini — APPROVED** [review]
lgtm

**Review by machacekondra — APPROVED** [review]

File: config/network/route.yaml (hunk 1 above)
Comment (arielireni) [reply config/network/route.yaml:10]: where should we pass the host here?
I tried to start yggdrasil with:
`sudo ./yggd --protocol http --path-prefix api/flotta-management/v1 --client-id $(cat /etc/machine-id) --cert-file /tmp/cert.pem --key-file /tmp/key.pem --ca-root /tmp/ca.pem --server 127.0.0.1:8043`
but received connection refused

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**